cypress/e2e/marc-authority/edit-marc-authority-save-keep-editing.cy.js

```diff
@@ -0,0 +1,242 @@
+        Logs.openFileDetails(marcFile.fileName);
+        for (let i = 0; i < marcFile.numOfRecords; i++) {
+          Logs.getCreatedItemsID(i).then((link) => {
+            createdAuthorityIDs.push(link.split('/')[5]);
+          });
+        }
+      });
+
+      cy.createTempUser([
+        Permissions.uiMarcAuthoritiesAuthorityRecordEdit.gui,
+        Permissions.uiMarcAuthoritiesAuthorityRecordView.gui,
+        Permissions.uiQuickMarcQuickMarcAuthoritiesEditorAll.gui,
+      ]).then((createdUserProperties) => {
+        testData.userProperties = createdUserProperties;
+        headerContent.editedHeaderContent.source.firstName = testData.userProperties.username;
+
+        cy.login(testData.userProperties.username, testData.userProperties.password);
+      });
+    });
+  });
+
+  beforeEach('Visit MARC Authorities', () => {
+    cy.visit(TopMenu.marcAuthorities);
+    MarcAuthorities.waitLoading();
+    MarcAuthorities.searchBy(testData.authority.searchOption, testData.authority.searchInput);
+    MarcAuthorities.select(`${createdAuthorityIDs[0]}${authorityPostfix}`);
+    MarcAuthority.edit();
+  });
+
+  after('Delete test data', () => {
+    cy.getAdminToken();
+    createdAuthorityIDs.forEach((id) => {
+      MarcAuthority.deleteViaAPI(id);
+    });
+    Users.deleteViaApi(testData.userProperties.userId);
+  });
+
+  it(
+    'C360092 Verify that click on the "Save & keep editing" button doesnt close the editing window of "MARC Authority" record (spitfire) (TaaS)',
+    { tags: [TestTypes.extendedPath, DevTeams.spitfire] },
+    () => {
+      // Verify initial state of edit view
+      QuickMarcEditor.checkHeaderFirstLine(
+        headerContent.initialHeaderContent.marcData,
+        headerContent.initialHeaderContent.source,
+      );
+      QuickMarcEditor.checkButtonsDisabled();
+
+      // Edit record and verify button states updated
+      MarcAuthority.changeField(
+        testData.editedFields[0].tag,
+        `${subfieldPrefix}${testData.editedFields[0].content}`,
+      );
+      QuickMarcEditor.checkButtonsEnabled();
+
+      // Save edits and verify view updated
+      QuickMarcEditor.clickSaveAndKeepEditing();
+      QuickMarcEditor.checkButtonsDisabled();
+      QuickMarcEditor.checkHeaderFirstLine(
+        headerContent.editedHeaderContent.marcData,
+        headerContent.editedHeaderContent.source,
+      );
+
+      // Add field and verify button states updated
+      MarcAuthority.addNewField(4, testData.newField.tag, testData.newField.content);
+      QuickMarcEditor.checkButtonsEnabled();
+
+      // Save added field and verify view updated
+      QuickMarcEditor.clickSaveAndKeepEditing();
+      QuickMarcEditor.checkButtonsDisabled();
+      QuickMarcEditor.checkHeaderFirstLine(
+        headerContent.editedHeaderContent.marcData,
+        headerContent.editedHeaderContent.source,
+      );
+
+      // Delete added field and verify states
+      cy.wait(4000);
```

Comment:
What this timeout for?

Reply:
it was used for waiting until Quick Marc Editor becomes interactive again after pressing "Save and Keep editing" button. I replaced it with an explicit waiter (assertion), thank you.